### PR TITLE
[recnet-api][v1.3.0] Add Invite Codes and Announcement endpoints

### PR DESCRIPTION
**Pull request by swh00tw**
## Description

Deploy Invite Code features (endpoints & UI) & Announcement feature (endpoint & new schemas).

PR:
- #236 
- #235 

## Related Issue

- #95 
- #63 
- #227 

## TODO

- [x] Paste the testing link: https://rec-net-git-recnet-api-dev-recnet-542617e7.vercel.app/feeds
- [x] Clear `console.log` or `console.error` for debug usage
- [x] Update the documentation `recnet-docs` if needed
- [x] Version bump in `package.json` if needed

### CODE REVIEW

File: apps/recnet-api/src/utils/pipes/zod.validation.pipe.ts [hunk 1]
```diff
@@ -4,13 +4,23 @@ import { ZodError, ZodSchema } from "zod";
...
 
   transform(value: unknown, metadata: ArgumentMetadata) {
     if (metadata.type === "custom") {
       return value;
     }
+
+    if (this.metadataType && metadata.type !== this.metadataType) {
+      return value;
+    }
+
```
Comment (swh00tw) [inline apps/recnet-api/src/utils/pipes/zod.validation.pipe.ts:23]: Why would we need this?

Comment (joannechen1223) [reply apps/recnet-api/src/utils/pipes/zod.validation.pipe.ts:23]: In `PATCH /announcement/:id` API, the body validator will be applied to parameter. As a result, I added this as a workaround to resolve this issue.

**Review by joannechen1223 — APPROVED** [review]
Please commit the version change in `package.json` before you merge to master.

File: apps/recnet-api/src/utils/pipes/zod.validation.pipe.ts (hunk 1 above)
Comment (swh00tw) [reply apps/recnet-api/src/utils/pipes/zod.validation.pipe.ts:23]: We will create another PR to make validationPipe API easier to use and understand.

**Pull request merged**